src/credentials/CHIPCert.h

```diff
@@ -53,8 +53,8 @@ static constexpr uint16_t kX509NoWellDefinedExpirationDateYear = 9999;
 static constexpr uint32_t kMaxCHIPCertLength = 400;
 static constexpr uint32_t kMaxDERCertLength  = 600;
 
-// The decode buffer is used to reconstruct TBS section of X.509 certificate, which doesn't include signature.
-static constexpr uint32_t kMaxCHIPCertDecodeBufLength = kMaxDERCertLength - Crypto::kMax_ECDSA_Signature_Length_Der;
+// As per spec section 11.24 (Wi-Fi Authentication with Per-Device Credentials)
+inline constexpr uint32_t kMaxCHIPCompactNetworkIdentityLength = 137;
```

Comment:
Please consider giving a bit of slack for unforeseen expansions.

Reply:
Adding anything optional to the cert (that older implementations can ignore) would break the ability to round-trip to the compact representation. It's also hard to think of anything we might want to add in that way; since these are self signed adding any data into the cert doesn't make that data any more trusted than passing along the same information outside of the cert itself. Effectively these are by design a minimal way of wrapping a key pair into a (EAP-)TLS-compatible cert. If we wanted to add anything fundamental (like supporting larger key sizes or different algorithms we're going to need a "v2" and both sides will need to be able to figure out that the other side supports it.